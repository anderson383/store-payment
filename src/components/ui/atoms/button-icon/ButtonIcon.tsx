import styles from './ButtonIcon.module.scss'
interface ButtonIconProps {
  icon: React.ReactNode
}
export const ButtonIcon:React.FC<ButtonIconProps> = ({icon}) => {
  return (
    <button className={styles.buttonIcon}>
      {icon}
    </button>
  )
}