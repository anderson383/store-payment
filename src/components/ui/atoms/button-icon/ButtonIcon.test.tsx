
import {
  fireEvent,
  render,
  screen
} from '@testing-library/react';
import { act } from 'react';
import { ButtonIcon } from './ButtonIcon';
import { IconArrowLeft } from '@tabler/icons-react';

describe('Tests in ButtonIcon component', () => {
  const setup = () => {
    return render(
      <ButtonIcon icon={<IconArrowLeft />}  />
    );
  };

  test('Should render the component', async () => {
    const {asFragment} = setup();
    
    await act(() => Promise.resolve())
    expect(asFragment()).toMatchSnapshot();
  });
});
