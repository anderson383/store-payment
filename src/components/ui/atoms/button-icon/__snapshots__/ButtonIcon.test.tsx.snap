// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in ButtonIcon component Should render the component 1`] = `
<DocumentFragment>
  <button
    class="buttonIcon"
  >
    <svg
      class="tabler-icon tabler-icon-arrow-left "
      fill="none"
      height="24"
      stroke="currentColor"
      stroke-linecap="round"
      stroke-linejoin="round"
      stroke-width="2"
      viewBox="0 0 24 24"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5 12l14 0"
      />
      <path
        d="M5 12l6 6"
      />
      <path
        d="M5 12l6 -6"
      />
    </svg>
  </button>
</DocumentFragment>
`;
